### ISSUE TITLE
Add "pinning" to navigation overlay

### ISSUE THREAD

**Issue by liuche** [mozilla-mobile/firefox-tv#441]
Add the "pin" UI to the navigation overlay, and save the pinned items and load them to the home screen.

See specs in #433.

**Strings** for confirmation toast:
"Pinned to Firefox TV homescreen"
"Removed from Firefox TV homescreen"

**SVGs**
[pin-filled](https://gist.github.com/liuche/fcfb3077680dd5499a5908d4e3f745e1)
[pin-unfilled](https://gist.github.com/liuche/43191558a607090cf193f07b35116a56)

**Comment by lime124** [comment]
@liuche and i discussed to not limit the number of tiles for this first version. I want to see how many tiles people are adding. Can we get telemetry in for this?

**Comment by Sdaswani** [comment]
@liuche @bkmunar @mcomella  is there any telemetry planned for adding a custom tile yet? Or do we need to create a new ticket? cc @bbinto

**Comment by bbinto** [comment]
Leaving this here for now, but we def. need telemetry :)

Some events/probes I can think of
- differentiate between custom tile and default tiles
- delete custom tile
- delete default tiles
- add custom tile
- count # of custom tiles, and total tiles (to create a histogram)

@lime124. @liuche would you mind posting some UI specs so I can see how the actually UI looks like to finalize the probes?

I'm not an engineer anymore but I'd feel we should have a max value though, even if it's high :)

**Comment by liuche** [comment]
@bbinto See the mocks in #433

**Comment by bbinto** [comment]
@liuche I was hoping to see the flow how the pinning works. Couldn't find it in #433??!?!

**Comment by mcomella** [comment]
@lime124 We also need strings for a11y, namely:
- When the user mouses over the pinned button, e.g. "Pin top site". If we use a "checkbox widget, this would become: "Pin top site: checkbox, checked" and "Pin top site: checkbox, not checked' for not pinned and pinned respectively. We could also implement the checking ourselves and use two strings (e.g. "pin" and "unpin") but that's more complicated in code and I'd probably not recommend it.
- When the user focus a custom pinned top site on the homescreen (so they can choose to unpin it if they want), e.g. "Top site, youtube.com, pinned top site".

**Comment by mcomella** [comment]
FYI: I'm going to extract the image and title in #442 because 1) it'll use the same mocks as the images and 2) it's likely going to use the same code path.

**Comment by mcomella** [comment]
> FYI: I'm going to extract the image and title in #442

I figured out what needs to happen and it's not blocked on #442 so I filed #511.

**Comment by lime124** [comment]
@mcomella i talked with eitan and he suggested we use the same pattern as the data usage checkbox in settings. basically turbo mode, checkbox, checked.

I don't believe this generates new strings to be translated as we'd already have turbo mode, checkbox, and checked from other stuff. We might need pin?

**Comment by liuche** [comment]
@lime124 ugh, I thought this svg was working (from the previews in Android Studio) but in the app it still looks wrong :(

Maybe something about the svg Non-Zero fills vs Even-Odd fills...? I know you made this in Illustrator rather than Sketch, but maybe it's this? https://medium.com/sketch-app-sources/exporting-svgs-in-sketch-for-android-169ee8dae603

<img width="485" alt="screen shot 2018-02-23 at 4 53 19 pm" src="https://user-images.githubusercontent.com/1082253/36623493-12fa4de8-18ba-11e8-9753-301177a896af.png">

**Comment by mcomella** [comment]
@liuche @lime124 Andrzej once wrote a post on some common SVG->VectorDrawable errors: maybe it can tell you why it looks wrong? https://www.ahunt.org/2017/02/fun-with-vectordrawables/

**Comment by lime124** [comment]
ugh sorry it seems like this comment didn't get posted for some reason

Friday Feb 23 @mcomella i talked with eitan and he suggested we use the same pattern as the data usage checkbox in settings. basically turbo mode, checkbox, checked.

I don't believe this generates new strings to be translated as we'd already have turbo mode, checkbox, and checked from other stuff. We might need pin?

**Comment by lime124** [comment]
@shorlander do you have any ideas? it's odd this doesn't seem to be a problem with any other icon we have. it's one that bryan gave me (twice now).

**Comment by liuche** [comment]
We also need an accessibility string for this, similar to how we have "Turbo mode" in the overlay: https://github.com/mozilla-mobile/firefox-tv/blob/master/app/src/main/res/values/strings.xml#L563

For turbo mode for example, the string is announced in voice view when the user highlights the turbo button in the overlay as "Turbo mode: checked" or "Turbo mode: unchecked". We'd need a similar string for Pinning. (Maybe something like..."Pin site" or "Pinned site" - but both of these are slightly confusing...)

**Comment by lime124** [comment]
@liuche @BrianNJones eitan suggested we do the same for pinning as turbo mode - call it the things then checkbox then checked. So I guess the big difference between the two would be "turbo mode" vs "pin"

**Comment by liuche** [comment]
Okay, so the string can be "Pin"? Just wanted to explicitly ask what string I should add as the label for the thumbtack icon.

**Comment by BrianNJones** [comment]
@liuche @lime124 I think we need a slightly more expansive accessibility string than "Pin." Recommend "Pin to homescreen"...is there concern that there may be confusion re: whether reference is to Firefox TV vs. device homescreen?

**Comment by liuche** [comment]
Note for QA: this doesn't handle the "bundled" sites yet, that is #533.

@lime124, if you want to build this and see how pinning feels, this has landed on master!
